cpp/src/arrow/csv/writer.cc

```diff
@@ -117,17 +116,16 @@ class ColumnPopulator {
   }
 
   // Places string data onto each row in output and updates the corresponding row
-  // pointers in preparation for calls to other (preceding) ColumnPopulators.
+  // pointers in preparation for calls to other (next) ColumnPopulators.
   // Implementations may apply certain checks e.g. for illegal values, which in case of
   // failure causes this function to return an error Status.
   // Args:
   //   output: character buffer to write to.
-  //   offsets: an array of end of row column within the the output buffer (values are
-  //   one past the end of the position to write to).
-  virtual Status PopulateColumns(char* output, int32_t* offsets) const = 0;
+  //   offsets: an array of start of row column within the output buffer.
+  virtual Status PopulateRows(char* output, int64_t* offsets) const = 0;
```

Comment:
Name `PopulateColumns` is a bit confusing IMO as this function writes rows of a single column to csv output buffer.

Reply:
The new name seems fine to me